### ISSUE TITLE
Unable to download a package with the latest version 

### ISSUE THREAD

**Issue by emimarz** [composer/composer#6580]
I create a fresh symfony
composer create-project symfony/framework-standard-edition test "2.4.*" 
I add some repository

My `composer.json`:

```json
{
    "name": "symfony/framework-standard-edition",
    "license": "MIT",
    "type": "project",
    "description": "The \"Symfony Standard Edition\" distribution",
    "autoload": {
        "psr-0": { "": "src/" }
    },
    "require": {
        "php": ">=5.3.3",
        "symfony/symfony": "2.4.*",
        "doctrine/orm": "~2.2,>=2.2.3",
        "doctrine/doctrine-bundle": "~1.2",
        "twig/extensions": "~1.0",
        "symfony/assetic-bundle": "~2.3",
        "symfony/swiftmailer-bundle": "~2.3",
        "symfony/monolog-bundle": "~2.4",
        "sensio/distribution-bundle": "~2.3",
        "sensio/framework-extra-bundle": "~3.0",
        "sensio/generator-bundle": "~2.3",
        "incenteev/composer-parameter-handler": "~2.0",
        "symfony/publisher": "dev-master"
    },
    "repositories": [
        {
            "type": "gitlab",
            "url": "http://gitlab.local/symfony/publisher.git"
        }
    ],
    "gitlab-domains": [
        "gitlab.local",
        "gitlab.com"
    ],
    "scripts": {
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile"
        ]
    },
    "config": {
        "bin-dir": "bin",
        "secure-http": false
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-web-dir": "web",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        },
        "branch-alias": {
            "dev-master": "2.4-dev"
        }
    }
}

```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
Defining autoload.psr-0 with an empty namespace prefix is a bad idea for performance
require.symfony/publisher : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK

```

When I run this command:

```
composer require symfony/publisher dev-master -vvv
```

I get the following output:

```
Reading ./composer.json
Loading config file /home/blabla/.composer/auth.json
Loading config file ./composer.json
Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
Executing command (/var/www/test): git branch --no-color --no-abbrev -v
Executing command (/var/www/test): git describe --exact-match --tags
Executing command (/var/www/test): git log --pretty="%H" -n1 HEAD
Executing command (/var/www/test): hg branch
Executing command (/var/www/test): fossil branch list
Executing command (/var/www/test): fossil tag list
Executing command (/var/www/test): svn info --xml
Failed to initialize global composer: Composer could not find the config file: /home/blabla/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /var/www/test/vendor/composer/installed.json
Running 1.4.2 (2017-05-17 08:17:52) with PHP 5.5.38-1~dotdeb+7.1 on Linux / 3.2.0-4-amd64
./composer.json has been updated
Reading ./composer.json
Loading config file /home/blabla/.composer/auth.json
Loading config file ./composer.json
Executing command (/var/www/test): git branch --no-color --no-abbrev -v
Executing command (/var/www/test): git describe --exact-match --tags
Executing command (/var/www/test): git log --pretty="%H" -n1 HEAD
Executing command (/var/www/test): hg branch
Executing command (/var/www/test): fossil branch list
Executing command (/var/www/test): fossil tag list
Executing command (/var/www/test): svn info --xml
Failed to initialize global composer: Composer could not find the config file: /home/blabla/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /var/www/test/vendor/composer/installed.json
Reading ./composer.lock
Loading composer repositories with package information
Downloading http://gitlab.local/api/v3/projects/symfony%2Fpublisher
Warning: Accessing gitlab.local over http which is an insecure protocol.
Failed to download symfony/publisher:The "http://gitlab.local/api/v3/projects/symfony%2Fpublisher" file could not be downloaded (HTTP/1.1 404 Not Found)
Your credentials are required to fetch private repository metadata (http://gitlab.local/symfony/publisher.git)
A token will be created and stored in "/home/blabla/.composer/auth.json", your password will never be stored
To revoke access to this token you can visit gitlab.local/profile/applications
Username: emimarz
Password: Downloading http://gitlab.local/oauth/token

Token successfully created
Downloading http://gitlab.local/api/v3/projects/symfony%2Fpublisher

Installation failed, reverting ./composer.json to its original content.

                                                                                                                                             
  [Composer\Downloader\TransportException]                                                                                                   
  The "http://gitlab.local/api/v3/projects/symfony%2Fpublisher" file could not be downloaded (HTTP/1.1 404 Not Found)  
                                                                                                                                             

Exception trace:
 () at phar:///usr/bin/composer/src/Composer/Util/RemoteFilesystem.php:365
 Composer\Util\RemoteFilesystem->get() at phar:///usr/bin/composer/src/Composer/Util/RemoteFilesystem.php:101
 Composer\Util\RemoteFilesystem->getContents() at phar:///usr/bin/composer/src/Composer/Repository/Vcs/VcsDriver.php:163
 Composer\Repository\Vcs\VcsDriver->getContents() at phar:///usr/bin/composer/src/Composer/Repository/Vcs/GitLabDriver.php:392
 Composer\Repository\Vcs\GitLabDriver->getContents() at phar:///usr/bin/composer/src/Composer/Repository/Vcs/GitLabDriver.php:295
 Composer\Repository\Vcs\GitLabDriver->fetchProject() at phar:///usr/bin/composer/src/Composer/Repository/Vcs/GitLabDriver.php:91
 Composer\Repository\Vcs\GitLabDriver->initialize() at phar:///usr/bin/composer/src/Composer/Repository/VcsRepository.php:88
 Composer\Repository\VcsRepository->getDriver() at phar:///usr/bin/composer/src/Composer/Repository/VcsRepository.php:123
 Composer\Repository\VcsRepository->initialize() at phar:///usr/bin/composer/src/Composer/Repository/ArrayRepository.php:185
 Composer\Repository\ArrayRepository->getPackages() at phar:///usr/bin/composer/src/Composer/DependencyResolver/Pool.php:104
 Composer\DependencyResolver\Pool->addRepository() at phar:///usr/bin/composer/src/Composer/Installer.php:376
 Composer\Installer->doInstall() at phar:///usr/bin/composer/src/Composer/Installer.php:223
 Composer\Installer->run() at phar:///usr/bin/composer/src/Composer/Command/RequireCommand.php:180
 Composer\Command\RequireCommand->execute() at phar:///usr/bin/composer/vendor/symfony/console/Command/Command.php:267
 Symfony\Component\Console\Command\Command->run() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:846
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:191
 Symfony\Component\Console\Application->doRun() at phar:///usr/bin/composer/src/Composer/Console/Application.php:227
 Composer\Console\Application->doRun() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:122
 Symfony\Component\Console\Application->run() at phar:///usr/bin/composer/src/Composer/Console/Application.php:100
 Composer\Console\Application->run() at phar:///usr/bin/composer/bin/composer:54
 require() at /usr/bin/composer:24

require [--dev] [--prefer-source] [--prefer-dist] [--no-progress] [--no-suggest] [--no-update] [--no-scripts] [--update-no-dev] [--update-with-dependencies] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--] [<packages>]...
```

And I expected this to happen:
that install the bundle

NOTE: after try install one time the name of the bundle remain in require section instead be deleted
I remove manually

**Comment by alcohol** [comment]
I think the error is rather obvious.

**Comment by emimarz** [comment]
@alcohol can you share?

**Comment by alcohol** [comment]
404.

**Comment by emimarz** [comment]
so what? Yes, it give a 404 but it shouldn't, I am using my credentials and in browser when I'm logged in I can access to that url. I don't know why this stop working. I need some help here, I'm stuck. Nobody touch my gitlab server.

**Comment by curry684** [comment]
```
Downloading http://gitlab.local/api/v3/projects/symfony%2Fpublisher
```
That URL-encoding is highly suspicious, that shouldn't be in logs like this. Could be a true bug.

Also, why is it connecting to API v3? That's a terribly bad idea by now: https://docs.gitlab.com/ce/api/v3_to_v4.html

**Comment by alcohol** [comment]
@curry684 I do not know if we actually support v4 already. Also, that url encoding is from our end and it has to be there afaik because the project name can contain a `/`, which is also a url delimiter.

**Comment by curry684** [comment]
Well it's going to break in a month so someone using it should likely make time for it   :P   We have GitLab in our office but we just use Satis for distribution.

**Comment by Seldaek** [comment]
Seems like a duplicate of #6453 which I fixed earlier, and it will be included in composer 1.5.0
